From 43784084e79d8701bb7f96fb6204e91a50e608df Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Thu, 25 Jan 2018 18:43:24 +0800
Subject: [PATCH 3299/5242] MLK-17473-5 drm/imx: core: allow fb modifiers for
 DCSS

commit  a0908ed8c09cf4e1d768760782117e57d6b82c80 from
https://source.codeaurora.org/external/imx/linux-imx.git

Set the 'allow_fb_modifiers' flag to be true when DCSS
exists to make the format modifiers blob data can be
created correctly during the plane initialization.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/imx-drm-core.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/imx-drm-core.c b/drivers/gpu/drm/imx/imx-drm-core.c
index 39255b9..6412a28 100644
--- a/drivers/gpu/drm/imx/imx-drm-core.c
+++ b/drivers/gpu/drm/imx/imx-drm-core.c
@@ -288,7 +288,7 @@ static int imx_drm_bind(struct device *dev)
 	drm->mode_config.max_height = 4096;
 	drm->mode_config.allow_fb_modifiers = true;
 
-	if (has_dpu(dev)) {
+	if (has_dpu(dev) || has_dcss(dev)) {
 		drm->mode_config.allow_fb_modifiers = true;
 		dev_dbg(dev, "allow fb modifiers\n");
 	}
-- 
1.7.9.5

